src/gpu/intel/jit/gemm/include/config.hpp

```diff
@@ -17,4 +17,111 @@
+template <ngen::HW hw>
+using PostOpInjector = dnnl::impl::gpu::intel::jit::post_op_injector_t<GENERATOR_BASE(hw)>;
+constexpr int maxPostOps = dnnl::impl::post_ops_t::post_ops_limit;
+
+static inline BinaryOp toBinaryOp(const PostOps::entry_t &e)
+{
+    using namespace dnnl::impl;
+    switch (e.as_binary().alg) {
+        case alg_kind::binary_add:   return BinaryOp::Add;
+        case alg_kind::binary_sub:   return BinaryOp::Sub;
+        case alg_kind::binary_mul:   return BinaryOp::Mul;
+        case alg_kind::binary_div:   return BinaryOp::Div;
+        case alg_kind::binary_min:   return BinaryOp::Min;
+        case alg_kind::binary_max:   return BinaryOp::Max;
+        case alg_kind::binary_prelu: return BinaryOp::Prelu;
+        default: gpu_error_not_expected();
+    }
+    return BinaryOp::Add;
+}
+
+template <ngen::HW hw>
+void injectNonBinaryPostOps(const PostOps::entry_t & entry, GENERATOR_BASE(hw) *g,
+                            ngen::RegisterAllocator ra, int C_grfs[ngen::GRF::maxRegs()],
+                            int C_ngrf, bool postOpFwd) {
+    namespace jit = dnnl::impl::gpu::intel::jit;
+    switch (entry.kind()) {
+        case dnnl::impl::gpu::intel::post_op::kind_t::eltwise: {
+            using Injector = jit::eltwise_injector_f32_t<jit::generator_t<hw>>;
+            int euCount = 0; /* only used for a DG2 W/A for conv */
+            auto &ee = entry.as_eltwise();
+            Injector injector{g, ee.alg, ee.alpha, ee.beta, ee.scale,
+                              euCount, ngen::GRFRange(), postOpFwd};
+
+            auto scratch = ra.try_alloc_range(injector.preferred_scratch_regs());
+            if (scratch.isInvalid())
+                scratch = ra.alloc_range(injector.min_scratch_regs());
+
+            injector.set_scratch(scratch);
+            injector.prepare();
+            injector.compute(C_grfs, C_ngrf);
+            break;
+        }
+        default: gpu_error_not_expected();
+    }
+}
+
+template <ngen::HW hw>
+void injectStochasticRound(GENERATOR_BASE(hw) *g,
+                            ngen::RegisterAllocator ra, int C_grfs[ngen::GRF::maxRegs()],
+                           int C_ngrf, bool postOpFwd, const ngen::Subregister &seed, ngen::DataType t) {
+    namespace jit = dnnl::impl::gpu::intel::jit;
+    using Injector = jit::eltwise_injector_f32_t<jit::generator_t<hw>>;
+    int euCount = 0; /* only used for a DG2 W/A for conv */
+    Injector injector{g, dnnl::impl::alg_kind::eltwise_stochastic_round, 0.0, 0.0, 1.0,
+                      euCount, ngen::GRFRange(), postOpFwd};
+    auto scratch = ra.try_alloc_range(injector.preferred_scratch_regs());
+    if (scratch.isInvalid())
+        scratch = ra.alloc_range(injector.min_scratch_regs());
+    if (scratch.isInvalid())
+        gpu_error_not_expected();
+
+    injector.set_scratch(scratch);
+    injector.prepare();
+    injector.compute(C_grfs, C_ngrf, seed.getBase(), seed.getOffset(), t);
+
+}
```

Comment:
Can we make the oneDNN `PostOps` implementation a separate class (defined in `src/gpu/intel/jit/gemm` somewhere) that implements the above three methods? I think that could clean up a lot of this, and avoid the need for pre_config.hpp.